### PR TITLE
ENH: control a random stream to be from either numpy or pytorch

### PR DESCRIPTION
**Pull request by ev-br**
Add a global switch to select the random stream origin: pytorch or numpy. If points to NumPy, `tnp.random.uniform` just calls `numpy.random.uniform` etc.

### CODE REVIEW

File: torch_np/random.py [hunk 1]
```diff
@@ -29,21 +30,44 @@
     "randint",
     "shuffle",
     "uniform",
+    "RANDOM_STREAM",
 ]
 
 
+RANDOM_STREAM = "pytorch"
```
Comment (lezcano) [inline torch_np/random.py:37]: Just make it a boolean variable. Something like `USE_NUMPY_RANDOM = False`

**Review by lezcano — APPROVED** [review]
Just one small comment, otherwise it LGTM

**Comment by lezcano** [comment]
You can also now use this flag in the E2E tests.

File: torch_np/random.py (hunk 1 above)
Comment (ev-br) [reply torch_np/random.py:37]: The string value here parallels `set_default_dtype`: https://github.com/Quansight-Labs/numpy_pytorch_interop/blob/main/torch_np/_dtypes.py#L346

Comment (lezcano) [reply torch_np/random.py:37]: I still think it's better to have a boolean flag for this, as we just have two values. In the other case we also have other possible values, so the enum is alright. Although I'm not the biggest fan of the enum sugar in the dtypes either, but that's a story for another day ajjaja

Comment (ev-br) [reply torch_np/random.py:37]: OK, will change. My experience is that almost all boolean flags (modulo a set of measure zero, that is) have to be changed to more than two state eventually. But here it's OK for now indeed, so let me change it and move on :-).

**Pull request merged**